workshop/nano1-vuex.md

```diff
@@ -0,0 +1,281 @@
+
+```js
+const store = new Vuex.Store({
+  state: {
+    games: []
+  },
+
+  getters: {
+  },
+
+  mutations: {
+  },
+
+  actions: {
+  }
+});
+
+export default store;
+```
+
+We will cover everything in here in depth later on, but for now lets focus on `state`.
+
+Vuex is a state management library, and so we need to declare our global state for our application.
+Think of `state` on our store like a globally available `data() {}` in a Vue component. 
+
+We will add a `games` array which will hold our data soon.
+
+Go back to `main.js` and import our store:
+```js
+import store from "./store"; 
+```
+
+And finally add it to our Vue app:
+
+```js
+new Vue({
+  el: "#app",
+  components: { App },
+  template: "<App/>",
+  store
+}); 
+```
+
+## 3. Step 2 - Fetching our remote data
+
+In this nano we will pretend as if we were fetching data from some remote API, and receiving the contents of `data.json` into our application.
+
+You will notice that we are already importing it `import data from "./data/data.json";`, so head to our `getFromAPI()` method and add the following code:
+
+```js
+const apiData = data; // Our "pretend" ajax call
+apiData.forEach(game => {
+  this.$store.dispatch("addGame", game);
+}); 
+```
+
+We are assigning our imported `data` into a constant as a pretend ajax call, and then we are using the
+js array method `forEach` to loop through the array. 
+
+What exactly then is `$store.dispatch`?
+
+Vuex has three different state management types of methods.
+1. **Getters**. For getting data from the store
+2. **Mutations**. For mutating/changing the data in the store
+3. **Actions**. Similar to mutations, but they *commit* the mutations, and can hold other logic.
+
+In this case, we are using an **action** via *dispatch* to add a game to our store.
+
+::: tip 💡
+*Why am I not mutating the data with a commit?* It is a best practice to only commit from within actions.  
+:::
+
+Let's write our action and mutation to store our newly fetched data.
+
+Head over to our `store.js` and add the following mutation and action:
+
+```js
+mutations: {
+  addGame(state, game) {
+    state.games.push(game);
+  }
+}
+
+actions: {
+  addGame(context, game) {
+    if (typeof game !== Object && game !== null) return;
+    
+    context.commit("addGame", game);
+  }
+}
+```
+
+In our action, we are making a micro validation for the data we will insert. This is just an example
+of what you can do inside this function, you could also make axios/ajax calls and commit data on the
+responses!
+
+`context.commit` will call our *addGame* **mutation** and pass the data after we validate it.
+
+Inside our mutation, we simply push the game into our array.
+
+## 4. Step 3 - Getting our data from the store
+If you run the application at this point nothing will happen, but our data has already been 
+added to our vuex store.
+
+In order to use our Vue development tools, first click on the `Open in new window` button on the
```

Comment:
We need to mention you need to have Vue DevTools extension installed in user's browser (maybe links to extensions for Chrome and FF?)